### ISSUE TITLE
TestCaseSource is ingored when different main namespace than the others

### ISSUE THREAD

**Issue by marbel82** [nunit/nunit#4599]
## Tools
Microsoft Visual Studio Community 2022 (64-bit) - Version 17.9.0 Preview 2.1
```xml
    <PackageReference Include="NUnit" Version="4.0.1" />
    <PackageReference Include="NUnit3TestAdapter" Version="4.5.0">
```
----
## Problem
I update NUnit into 4.0.1 version.
**Some tests with TestCaseSource are ignored now.**

In the test project, I use the name style with "Tests" added at the end of class namespace. For example:

In project MyProject:
```csharp
namespace MyProject.Converters {}
namespace MyProject.Common {}
namespace MyProject.Model {}
namespace MyProject.Validators {}
```

In project MyProjectTests:
```csharp
namespace MyProject.Converters.Tests {}
namespace MyProject.Common.Tests {}
namespace MyProject.Model.Tests {}
namespace MyProjectTests.Validators {} // (default by creator)
```

In one file I forgot to change the namespace from the default one. The file with `namespace MyProjectTests.Validators` and `TestCaseSource` are skipped.

After changing the namespace into `MyProject.Validators.Tests`, `TestCaseSource` started working.

**Comment by manfred-brands** [comment]
What are the parameters for your `TestCaseSource`?
Is it a local member of your test class or do you reference an other class and if so what is the namespace of that class.

**Comment by marbel82** [comment]
I found the cause. The problem could have been for half a year. **As usual, the problem is with a static variable.**
Someone wrote the test and then someone changed the implementation.

The test looks something like this:

```csharp
[TestCaseSource(nameof(Documents))]
public void CheckDocumentType(DocumentType type)
{
    Assert.That(...);
}

private static IEnumerable<DocumentType> Documents()
{
    foreach (DocumentType type in DocumentType.StaticListOfDocuments)
    {
        yield return type;
    }
}
```
`DocumentType.StaticListOfDocuments` is now initialized by some method (for example `InitStaticListOfDocuments()`), and is not initialized in the `Documents()` method, from where test cases are taken.

When I changed the namespace, the order of tests has changed and some other test method call the `InitStaticListOfDocuments()` and static `DocumentType.StaticListOfDocuments` variable was initiated.
----

Why does the TestCaseSource not inform about exceptions?
